src/api/app/controllers/webui/webui_controller.rb

```diff
@@ -11,7 +11,8 @@ class Webui::WebuiController < ActionController::Base
 
   before_filter :setup_view_path
   before_filter :instantiate_controller_and_action_names
-  before_filter :set_return_to, :reset_activexml, :authenticate
+  before_filter :set_return_to, except: [:do_login, :login]
```

Comment:
I think that's ok. It sets the redirect for those actions but the redirect would only be used after calling /user/login or /user/do_login.
Of course there is some overhead of writing the session, but I am not sure if I want to exclude all the actions were this isn't necessary.